### PR TITLE
TEAMMATES logo displaced #6746

### LINKED ISSUE #6746: TEAMMATES logo displaced
This is the current view on the live server when logged in:

screen shot 2017-03-07 at 12 13 14 am

### PR DESCRIPTION
**Pull request by Dartteon**
Fixes #6746

**Outline of Solution**
Landing page (staticPage) uses the TEAMMATES logo for the home button in the navBar .tag file.
```
<a class="navbar-brand" href="/index.jsp">
    <img height="100%" src="/images/teammateslogo.jpg" alt="TEAMMATES[Logo] - Online Peer Feedback/Evaluation System for Student Team Projects">
</a>
```
The other .tag pages(/admin, /errorPage, /helpPage, /instructor, /student) are using a string literal for the same <a> component.
```
<a class="navbar-brand" href="/index.jsp">TEAMMATES</a>
```

**What I have done**
- Made a `navBarBrand.tag` to contain the brand element, so image can easily be edited in future
- For each of the .tag files of the various pages, I changed the navBar component to use the `navBarBrand.tag` instead of string "TEAMMATES"

**Before**
image

**After**
image

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/helpPage.tag [hunk 1]
```diff
@@ -31,9 +31,7 @@
     <div id="footerComponent" class="container-fluid">
         <div class="container">
             <div class="row">
-                <div class="col-md-2">
-                    <span>[<a href="/index.jsp">TEAMMATES</a>]</span>
-                </div>
+                <t:navBarBrand/>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/helpPage.tag]: Wrong place - you're editing the footer here, not the navbar. Also, same as above - only replace the existing `<a>` tag and its contents.

File: src/main/webapp/WEB-INF/tags/errorPage.tag [hunk 2]
```diff
@@ -20,9 +20,7 @@
 <body>
     <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
         <div class="container">
-            <div class="navbar-header">
-                <a class="navbar-brand" href="/index.jsp">TEAMMATES</a>
-            </div>
+            <t:navBarBrand/>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/errorPage.tag]: Wrong place - `<t:navBarBrand/>` should only replace the `<a>` tag (and its contents). You're changing the structure by discarding `<div class="navbar-header">`.

**Comment by whipermr5** [comment]
Also, for future reference, commit messages should be `Fix` not `Fixed`, `Make` not `Made` or `Making` (i.e. in the imperative mood - more about commit messages [here](https://oss-generic.github.io/process/docs/FormatsAndConventions.html)). Don't worry; I made this same mistake in [my first PR](https://github.com/TEAMMATES/teammates/pull/6342#issuecomment-272376270) too haha.

File: src/main/webapp/WEB-INF/tags/helpPage.tag [hunk 3]
```diff
@@ -31,7 +30,9 @@
     <div id="footerComponent" class="container-fluid">
         <div class="container">
             <div class="row">
-                <t:navBarBrand/>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/helpPage.tag]: This can be in line 23 instead.

Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/helpPage.tag]: Aren't the two of them different components? Line ~34 is for the footer, while Line 23 is for the header? O_o

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/helpPage.tag]: Yup; we're fixing the header not the footer right?

Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/helpPage.tag]: Woops, you're right. Will fix this, thanks! :)

**Comment by damithc** [comment]
The code looks good. However, I'll leave merge approval to @wkurniawan07 as I'm not familiar with this part of the code.

File: src/main/webapp/WEB-INF/tags/navBarBrand.tag [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<%@ tag description="TEAMMATES logo brand"%>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn" %>
+<%@ tag import="teammates.common.util.Const" %>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: Just noticed: this may not be necessary (does it work without)? The previous two lines it above too.

Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: Hm, you're absolutely right. I have ran the tests after removing the unnecessary lines and found that it works correctly, so I will be removing them!

**Comment by whipermr5** [comment]
@Dartteon The commit messages should be imperative, i.e. like you're giving instructions rather than describing what you did (so `Add` not `Adds`).

**Comment by Dartteon** [comment]
Oh boy, I thought I got it right by not using past-tenses already. Will take additional note in future, thanks! :)

File: src/main/webapp/WEB-INF/tags/navBarBrand.tag [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<%@ tag description="TEAMMATES logo brand"%>
+
+<a class="navbar-brand" href="/">
+    <img height="100%" src="../images/teammateslogo.jpg" alt="TEAMMATES[Logo] - Online Peer Feedback/Evaluation System for Student Team Projects">
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: Does the source need to be `../images/teammateslogo.jpg`? Will simply `images/teammateslogo.jpg` work?

File: src/main/webapp/WEB-INF/tags/navBarBrand.tag [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<%@ tag description="TEAMMATES logo brand"%>
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: `navBarBrand` is a class name taken from Bootstrap, right? It doesn't really tell the purpose of the tag file. `teammatesLogo.tag` is a better name.

Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: `navBarBrand` is a name @whipermr5 suggested in a previous PR. I will rename it then.

File: src/main/webapp/WEB-INF/tags/navBarBrand.tag (hunk 5 above)
Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: Yep. Somehow, when I use `images/teammatesLogo.jpg`, only the home page (staticPage) will display the image correctly.

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: The home page and the action pages have different request URLs, so the relative path to the image would be different - the home page should use `images/teammateslogo.jpg` while the other pages should use `../images/teammateslogo.jpg`. Now `../images` works for the home page only because one level up from the root still resolves to the root. One workaround is to use `/images/teammateslogo.jpg` instead, which is root-relative and so would be correct for all pages.

**Comment by Dartteon** [comment]
This is weird, TravisCI test has failed even though the tests ran correctly on my side. It seems that the cause of failure is also not something that I have changed?

**Comment by whipermr5** [comment]
Yep it's not your fault, but a network error on Travis' side.

File: src/main/webapp/WEB-INF/tags/teammatesLogo.tag [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<%@ tag description="TEAMMATES logo brand"%>
+
+<a class="navbar-brand" href="/">
+    <img height="100%" src="../images/teammateslogo.jpg" alt="TEAMMATES[Logo] - Online Peer Feedback/Evaluation System for Student Team Projects">
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/teammatesLogo.tag]: Could this be changed to the [root-relative URL](https://github.com/TEAMMATES/teammates/pull/6836#discussion_r106812391) mentioned above?

File: src/main/webapp/WEB-INF/tags/navBarBrand.tag (hunk 5 above)
Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/navBarBrand.tag]: Oh, that makes sense then. Will do this fix, thanks @whipermr5 !

File: src/main/webapp/WEB-INF/tags/teammatesLogo.tag (hunk 7 above)
Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/teammatesLogo.tag]: Yup! Just made a push, Travis should be checking now!

Comment (Dartteon) [inline src/main/webapp/WEB-INF/tags/teammatesLogo.tag]: The test is failing because I haven't updated the test HTML files with GodMode. Will do so in a bit.

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/teammatesLogo.tag]: A few more HTML files to update for the build to pass :)

File: src/test/resources/pages/instructorCommentsPageSearchNormal.html [hunk 8]
```diff
@@ -126,7 +126,7 @@
                   </span>
                   .instructor1@course1.tmt at ${datetime.now})
                 </span>
-                <a class="btn btn-default btn-xs icon-button pull-right" data-original-title="Edit comment in the Comments page" data-placement="top" data-toggle="tooltip" href="/page/instructorCommentsPage?user=comments.idOfInstructor1OfCourse1&courseid=comments.idOfTypicalCourse1#${comment.id}" rel="noopener noreferrer" style="display:none;" target="_blank" title="" type="button">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorCommentsPageSearchNormal.html]: This looks like an unrelated change

File: src/test/resources/pages/instructorFeedbackResultsPageRankRGQView.html [hunk 9]
```diff
@@ -414,13 +414,13 @@
                             </li>
                           </ul>
                         </div>
-                        <button class="btn btn-default btn-xs icon-button pull-right" data-original-title="Add comment" data-placement="top" data-toggle="tooltip" id="button_add_comment" onclick="showResponseCommentAddForm(0,1,1, { sectionIndex: 1 })" title="" type="button">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorFeedbackResultsPageRankRGQView.html]: This file is full of unrelated changes

File: src/test/resources/pages/instructorSearchPageSearchComments.html [hunk 10]
```diff
@@ -155,7 +155,7 @@
                   </b>
                   [Wed, 02 May 2012, 08:50 AM UTC]
                 </span>
-                <a class="btn btn-default btn-xs icon-button pull-right" data-original-title="Edit comment in the Comments page" data-placement="top" data-toggle="tooltip" href="/page/instructorCommentsPage?user=searchUI.idOfInstructor1OfCourse1&courseid=searchUI.idOfTypicalCourse1#${comment.id}" rel="noopener noreferrer" style="display:none;" target="_blank" title="" type="button">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchComments.html]: Here too

File: src/test/resources/pages/instructorCommentsPageSearchNormal.html [hunk 11]
```diff
@@ -126,7 +126,7 @@
...
+				<a class="btn btn-default btn-xs icon-button pull-right" data-original-title="Edit comment in the Comments page" data-placement="top" data-toggle="tooltip" href="/page/instructorCommentsPage?user=comments.idOfInstructor1OfCourse1&courseid=comments.idOfTypicalCourse1#${comment.id}" rel="noopener noreferrer" style="display:none;" target="_blank" title="" type="button">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorCommentsPageSearchNormal.html]: Hate to keep nitpicking, but the original file had 16 spaces, not 4 tabs 😅

Comment (Dartteon) [inline src/test/resources/pages/instructorCommentsPageSearchNormal.html]: That's not nitpicking, its a valid concern! I can't believe I missed this even after checking each line for space instead of tabs... :(

**Comment by whipermr5** [comment]
Yup well done @Dartteon! @wkurniawan07 Your [merge approval](https://github.com/TEAMMATES/teammates/pull/6836#issuecomment-287287108) please :)

**Comment by Dartteon** [comment]
Thank you all for assisting me on my first (soon-to-be) successful PR on a public repo! 😂

**Comment by wkurniawan07** [comment]
Oops, didn't notice that I'm supposed to give approval 😅
